src/m365/pp/commands/tenant/tenant-settings-set.spec.ts

```diff
@@ -0,0 +1,438 @@
+  });
+
+  it('passes validation when the disableMakerMatch option is set to false', async () => {
+    const actual = await command.validate({ options: { disableMakerMatch: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableMakerMatch option is set to true', async () => {
+    const actual = await command.validate({ options: { disableMakerMatch: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disablePreferredDataLocationForTeamsEnvironment option is set to false', async () => {
+    const actual = await command.validate({ options: { disablePreferredDataLocationForTeamsEnvironment: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disablePreferredDataLocationForTeamsEnvironment option is set to true', async () => {
+    const actual = await command.validate({ options: { disablePreferredDataLocationForTeamsEnvironment: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableAdminDigest option is set to false', async () => {
+    const actual = await command.validate({ options: { disableAdminDigest: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableAdminDigest option is set to true', async () => {
+    const actual = await command.validate({ options: { disableAdminDigest: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableDeveloperEnvironmentCreationByNonAdminUsers option is set to false', async () => {
+    const actual = await command.validate({ options: { disableDeveloperEnvironmentCreationByNonAdminUsers: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableDeveloperEnvironmentCreationByNonAdminUsers option is set to true', async () => {
+    const actual = await command.validate({ options: { disableDeveloperEnvironmentCreationByNonAdminUsers: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableBillingPolicyCreationByNonAdminUsers option is set to false', async () => {
+    const actual = await command.validate({ options: { disableBillingPolicyCreationByNonAdminUsers: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableBillingPolicyCreationByNonAdminUsers option is set to true', async () => {
+    const actual = await command.validate({ options: { disableBillingPolicyCreationByNonAdminUsers: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if the storageCapacityConsumptionWarningThreshold is not a valid number', async () => {
+    const actual = await command.validate({ options: { storageCapacityConsumptionWarningThreshold: 'foo' } }, commandInfo);
```

Comment:
Also for this one, let's add an extra test for a negative number and float number.